src/containers/MainPlotContainer/index.tsx

```diff
@@ -116,6 +131,40 @@ class MainPlotContainer extends React.Component<MainPlotContainerProps> {
         });
     }
 
+    private updateThumbnails(cellId: string, url: string | null) {
+        this.setState((state) => {
+            // Move the updated thumbnail to the front of the map entries
+            const mapCopy = new Map(state.cellIdToZarrThumbnailUrl);
+            mapCopy.delete(cellId);
+            const mapEntries: [string, string | null][] = [
+                [cellId, url],
+                ...Array.from(mapCopy.entries()),
+            ];
```

Comment:
Is it necessary to copy `cellIdToZarrThumbnailUrl` here? I know React is pretty militant (rightly) about keeping state updates immutable, but it might be okay in this case given that you're about to replace `cellIdToZarrThumbnailUrl` below anyways.